keras/backend/mxnet_backend.py

```diff
@@ -1907,7 +1907,18 @@ def arange(start, stop=None, step=1, dtype='int32'):
     The default type of the returned tensor is `'int32'` to
     match TensorFlow's default.
     """
+    #Match behavior of numpy by return empty sequence
     dtype = np.dtype(dtype)
+    if stop is None:
+        stop = start
+        start = 0
+    if (start - stop) * step >= 0:
+        return np.arange(0, dtype=dtype)
+    #Work around for single element list. In this case, if KerasSymbol is passed to eval,
+    #a scalar will be returned instead of array
+    #Also workaround for 'int64' dtype
+    if np.absolute(start - stop) <= np.absolute(step) or dtype == np.dtype('int64'):
+        return np.arange(start, stop, step, dtype=dtype)
```

Comment:
This is a work around for current eval logic:
```
    if isinstance(x, KerasSymbol):
        if hasattr(x, 'tensor'):
            if x.name in x.get_bind_values() and _MODEL is not None:
                _MODEL._sync_weights()
            ret = x.tensor.asnumpy()
        else:
            bind_values = dfs_get_bind_values(x)
            executor = x.symbol.simple_bind(mx.cpu(), grad_req='null')
            for v in executor.arg_dict:
                bind_values[v].copyto(executor.arg_dict[v])
            outputs = executor.forward(is_train=_LEARNING_PHASE)
            ret = outputs[0].asnumpy()

        if ret.shape == (1,):
            return ret[0]
```
To return an array instead of scalar when the list length is 1.